core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -783,8 +758,12 @@ protected final void writerSegmentStats(SegmentsStats stats) {
     }
 
     @Override
-    public long indexWriterRAMBytesUsed() {
-        return indexWriter.ramBytesUsed();
+    public long indexBufferRAMBytesUsed() {
+        if (refreshing) {
+            return 0;
+        } else {
+            return indexWriter.ramBytesUsed() + versionMap.ramBytesUsedForRefresh();
```

Comment:
watch out this can throw a `AlreadyClosedException`! I really wonder why we need the `refreshing` invariant... can you explain and  add a comment?